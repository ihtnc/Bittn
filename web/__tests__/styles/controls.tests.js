import React from 'react';
import renderer from 'react-test-renderer';

import {
  Wrapper,
  FieldWrapper,
  PanelWrapper,
  ListPanelWrapper,
  MarkerInfoWrapper,
  Header,
  MarkerInfoHeader,
  MarkerInfoField,
  ExpandableButtonWrapper,
  Status,
  Label,
  ErrorLabel,
  SubLabel,
  LevelLabel,
  Input,
  List,
  ListItem,
  Link,
  Button,
  SelectedIcon,
  ErrorIcon,
  IconButton,
  Table,
  Row,
  Column
} from "@styles/controls";

const shallow = (comp) => renderer.create(comp).toJSON();

describe("Controls styled components", () => {
  test("Wrapper should match latest snapshot", () => {
    const wrapper = shallow(<Wrapper />)
    expect(wrapper).toMatchSnapshot();
  });

  test("FieldWrapper should match latest snapshot", () => {
    const wrapper = shallow(<FieldWrapper />)
    expect(wrapper).toMatchSnapshot();
  });

  test("MarkerInfoWrapper should match latest snapshot", () => {
    const wrapper = shallow(<MarkerInfoWrapper />)
    expect(wrapper).toMatchSnapshot();
  });

  describe("ExpandableButtonWrapper", () => {
    test("should render correctly when disabled", () => {
      const wrapper = shallow(<ExpandableButtonWrapper disabled={true} />);
      expect(wrapper).toMatchSnapshot();
    });

    test("should render correctly when not disabled", () => {
      const wrapper = shallow(<ExpandableButtonWrapper />);
      expect(wrapper).toMatchSnapshot();
    });
  });

  test("Header should match latest snapshot", () => {
    const wrapper = shallow(<Header />)
    expect(wrapper).toMatchSnapshot();
  });

  test("MarkerInfoHeader should match latest snapshot", () => {
    const wrapper = shallow(<MarkerInfoHeader />)
    expect(wrapper).toMatchSnapshot();
  });

  test("MarkerInfoField should match latest snapshot", () => {
    const wrapper = shallow(<MarkerInfoField />)
    expect(wrapper).toMatchSnapshot();
  });

  test("Status should match latest snapshot", () => {
    const wrapper = shallow(<Status />)
    expect(wrapper).toMatchSnapshot();
  });

  test("Label should match latest snapshot", () => {
    const wrapper = shallow(<Label />)
    expect(wrapper).toMatchSnapshot();
  });

  test("ErrorLabel should match latest snapshot", () => {
    const wrapper = shallow(<ErrorLabel />)
    expect(wrapper).toMatchSnapshot();
  });

  test("SubLabel should match latest snapshot", () => {
    const wrapper = shallow(<SubLabel />)
    expect(wrapper).toMatchSnapshot();
  });

  test("Input should match latest snapshot", () => {
    const wrapper = shallow(<Input />)
    expect(wrapper).toMatchSnapshot();
  });

  test("Link should match latest snapshot", () => {
    const size = '18px';
    const wrapper = shallow(<Link />)
    expect(wrapper).toMatchSnapshot();
  });

  describe("Button", () => {
    test("should render correctly when disabled", () => {
      const icon = "background-image";
      const wrapper = shallow(<Button disabled={true} icon={icon} />);
      expect(wrapper).toMatchSnapshot();
    });

    test("should render correctly when not disabled", () => {
      const icon = "background-image"
      const wrapper = shallow(<Button icon={icon} />);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe("SelectedIcon", () => {
    test("should render correctly when selected", () => {
      const wrapper = shallow(<SelectedIcon selected={true} />);
      expect(wrapper).toMatchSnapshot();
    });

    test("should render correctly when not selected", () => {
      const wrapper = shallow(<SelectedIcon />);
      expect(wrapper).toMatchSnapshot();
    });
  });

  test("ErrorIcon should match latest snapshot", () => {
    const wrapper = shallow(<ErrorIcon />)
    expect(wrapper).toMatchSnapshot();
  });


  test("IconButton should match latest snapshot", () => {
    const icon = 'background-image';
    const wrapper = shallow(<IconButton icon={icon} />)
    expect(wrapper).toMatchSnapshot();
  });

  test("List should match latest snapshot", () => {
    const wrapper = shallow(<List />)
    expect(wrapper).toMatchSnapshot();
  });

  test("ListItem should match latest snapshot", () => {
    const wrapper = shallow(<ListItem />)
    expect(wrapper).toMatchSnapshot();
  });

  test("ListPanelWrapper should match latest snapshot", () => {
    const wrapper = shallow(<ListPanelWrapper />)
    expect(wrapper).toMatchSnapshot();
  });

  describe("LevelLabel", () => {
    test("should render correctly for above level", () => {
      const wrapper = shallow(<LevelLabel level={1} />);
      expect(wrapper).toMatchSnapshot();
    });

    test("should render correctly for same level", () => {
      const wrapper = shallow(<LevelLabel level={0} />);
      expect(wrapper).toMatchSnapshot();
    });

    test("should render correctly for below level", () => {
      const wrapper = shallow(<LevelLabel level={-1} />);
      expect(wrapper).toMatchSnapshot();
    });

    test("should render correctly for default level", () => {
      const wrapper = shallow(<LevelLabel />);
      expect(wrapper).toMatchSnapshot();
    });
  });

  test("PanelWrapper should match latest snapshot", () => {
    const wrapper = shallow(<PanelWrapper />)
    expect(wrapper).toMatchSnapshot();
  });

  test("Table should match latest snapshot", () => {
    const wrapper = shallow(<Table />)
    expect(wrapper).toMatchSnapshot();
  });

  test("Row should match latest snapshot", () => {
    const wrapper = shallow(<Row />)
    expect(wrapper).toMatchSnapshot();
  });

  test("Column should match latest snapshot", () => {
    const wrapper = shallow(<Column />)
    expect(wrapper).toMatchSnapshot();
  });
});