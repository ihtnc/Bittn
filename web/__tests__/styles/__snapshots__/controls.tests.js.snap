// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls styled components Button should render correctly when disabled 1`] = `
<span
  className="css-1bz3hzy-Button e1ehg20213"
  disabled={true}
/>
`;

exports[`Controls styled components Button should render correctly when not disabled 1`] = `
<span
  className="css-e5ocwc-Button e1ehg20213"
/>
`;

exports[`Controls styled components Column should match latest snapshot 1`] = `
<td
  className="css-1x554m-Column e1ehg20224"
/>
`;

exports[`Controls styled components ErrorIcon should match latest snapshot 1`] = `
<span
  className="css-1bs8t5j-ErrorIcon e1ehg20215"
/>
`;

exports[`Controls styled components ErrorLabel should match latest snapshot 1`] = `
<div
  className="css-2ri7jk-ErrorLabel e1ehg2029"
/>
`;

exports[`Controls styled components ExpandableButtonWrapper should render correctly when disabled 1`] = `
<span
  className="css-ie68j7-ExpandableButtonWrapper e1ehg2023"
  disabled={true}
/>
`;

exports[`Controls styled components ExpandableButtonWrapper should render correctly when not disabled 1`] = `
<span
  className="css-1kefsjr-ExpandableButtonWrapper e1ehg2023"
/>
`;

exports[`Controls styled components FieldWrapper should match latest snapshot 1`] = `
<div
  className="css-12lwwuy-FieldWrapper e1ehg2021"
/>
`;

exports[`Controls styled components Header should match latest snapshot 1`] = `
<div
  className="css-1yo68oz-Header e1ehg2024"
/>
`;

exports[`Controls styled components IconButton should match latest snapshot 1`] = `
<span
  className="css-q92v05-IconButton e1ehg20216"
/>
`;

exports[`Controls styled components Input should match latest snapshot 1`] = `
<input
  className="css-51f26o-Input e1ehg20211"
/>
`;

exports[`Controls styled components Label should match latest snapshot 1`] = `
<div
  className="css-74rito-Label e1ehg2028"
/>
`;

exports[`Controls styled components LevelLabel should render correctly for above level 1`] = `
<span
  className="css-tvjils-LevelLabel e1ehg20220"
/>
`;

exports[`Controls styled components LevelLabel should render correctly for below level 1`] = `
<span
  className="css-q198i1-LevelLabel e1ehg20220"
/>
`;

exports[`Controls styled components LevelLabel should render correctly for default level 1`] = `
<span
  className="css-q198i1-LevelLabel e1ehg20220"
/>
`;

exports[`Controls styled components LevelLabel should render correctly for same level 1`] = `
<span
  className="css-ofqk23-LevelLabel e1ehg20220"
/>
`;

exports[`Controls styled components Link should match latest snapshot 1`] = `
<span
  className="css-1wkv0l-Link e1ehg20212"
/>
`;

exports[`Controls styled components List should match latest snapshot 1`] = `
<ol
  className="css-30fw97-List e1ehg20217"
/>
`;

exports[`Controls styled components ListItem should match latest snapshot 1`] = `
<ul
  className="css-1jjsim5-ListItem e1ehg20218"
/>
`;

exports[`Controls styled components ListPanelWrapper should match latest snapshot 1`] = `
<div
  className="css-1pb1k25-ListPanelWrapper e1ehg20219"
/>
`;

exports[`Controls styled components MarkerInfoField should match latest snapshot 1`] = `
<div
  className="css-ntsi7y-MarkerInfoField e1ehg2026"
/>
`;

exports[`Controls styled components MarkerInfoHeader should match latest snapshot 1`] = `
<div
  className="css-1g0s5le-MarkerInfoHeader e1ehg2025"
/>
`;

exports[`Controls styled components MarkerInfoWrapper should match latest snapshot 1`] = `
<div
  className="css-emkzpi-MarkerInfoWrapper e1ehg2022"
/>
`;

exports[`Controls styled components PanelWrapper should match latest snapshot 1`] = `
<div
  className="css-1l680q6-PanelWrapper e1ehg20221"
/>
`;

exports[`Controls styled components Row should match latest snapshot 1`] = `
<tr
  className="css-5pngix-Row e1ehg20223"
/>
`;

exports[`Controls styled components SelectedIcon should render correctly when not selected 1`] = `
<span
  className="css-1227m3b-SelectedIcon e1ehg20214"
/>
`;

exports[`Controls styled components SelectedIcon should render correctly when selected 1`] = `
<span
  className="css-r47jh3-SelectedIcon e1ehg20214"
  selected={true}
/>
`;

exports[`Controls styled components Status should match latest snapshot 1`] = `
<div
  className="css-d51yz3-Status e1ehg2027"
/>
`;

exports[`Controls styled components SubLabel should match latest snapshot 1`] = `
<div
  className="css-1mo7jhl-SubLabel e1ehg20210"
/>
`;

exports[`Controls styled components Table should match latest snapshot 1`] = `
<table
  className="css-1tb66d1-Table e1ehg20222"
/>
`;

exports[`Controls styled components Wrapper should match latest snapshot 1`] = `
<div
  className="css-smrfxc-Wrapper e1ehg2020"
/>
`;
